### PR TITLE
stk_mesh: Fix for changed get_updated_ngp_mesh API

### PR DESCRIPTION
**Pull request by ldh4**
**Pull-request type:**
- [ ] Bug fix 
- [ ] Documentation update
- [ ] Feature enhancement

## Description of the pull-request

get_updated_ngp_mesh() in STK BulkData will be deprecated.

stk::mesh::get_updated_ngp_mesh(BulkData) should be used instead.

## Checklist

**NOTE** The checklist below is a suggestion and not mandatory. You don't have
to _check all boxes_ to submit a pull request. However, please add a brief
explanation in your pull request summary explaining the omission for the benefit
of reviewers and developers.

*All pull requests*
- [x] Builds successfully (*must test on at least one system/compiler combination*)
  - Operating systems 
    - [x] Linux
    - [ ] MacOS
  - Compilers 
    - [x] GCC
    - [ ] LLVM/Clang
    - [ ] Intel compilers
    - [x] NVIDIA CUDA
- [x] Compiles without warnings
- [x] Passes all unit tests
- [ ] Passes all regression tests
- [ ] Documentation builds without errors

*For new code updates*
- [ ] Documentation updates - additions to user, theory, and verification manuals
- [ ] New unit tests providing coverage for new code, bug fixes etc.
- [ ] New regression tests exercising the new code

### CODE REVIEW

**Comment by sayerhs** [comment]
@ldh4 I would like to wait to merge this until we have had a chance to update trilinos all the production systems.

File: nalu.C [hunk 1]
```diff
@@ -25,6 +25,7 @@
 // input params
 #include <stk_util/environment/OptionsSpecification.hpp> 
 #include <stk_util/environment/ParseCommandLineArgs.hpp> 
+#include <stk_util/environment/ParsedOptions.hpp>
```
Comment (alanw0) [inline nalu.C:28]: Why was this include added?

**Comment by alanw0** [comment]
@sayerhs It's fine to wait until trilinos is updated on other platforms.
In fact we'll be adding a follow-up stk update to trilinos either today or tomorrow to fix some minor things that the trilinos builds found, which the sierra builds hadn't found.

**Comment by ldh4** [comment]
@sayerhs I agree that this merge should wait until updated Trilinos is in place. How often is it updated?

File: nalu.C (hunk 1 above)
Comment (ldh4) [reply nalu.C:28]: We found out that building Nalu with the latest Trilinos build (with latest snapshot of STK) causes incomplete type compile error on `ParsedOption` vars.
It seemed that `ParsedOptions.hpp` was previously included by `ParseCommandLineArgs.hpp` but was removed recently.

Comment (alanw0) [reply nalu.C:28]: ok, thanks for the explanation.

**Comment by jrood-nrel** [comment]
I tried this PR with the latest Trilinos and I see this error:
```
..../nalu-wind/include/ngp_utils/NgpMeshInfo.h:20:10: fatal error: stk_mesh/base/GetNgpMesh.hpp: No such file or directory
 #include "stk_mesh/base/GetNgpMesh.hpp"
```

**Comment by alanw0** [comment]
@jrood-nrel It looks like the develop branch in Trilinos that has our latest stk snapshot, hasn't been promoted to the master branch yet.

**Comment by jrood-nrel** [comment]
Oh I didn't realize my test picked up the master branch, you're right. This works with the develop branch. I'm fine merging it. I guess Trilinos needs to be updated in the container here?

**Comment by alanw0** [comment]
@jrood-nrel Yes Trilinos needs to be updated before that PR is merged. If you're fine with the develop branch go ahead, but if you want to wait til it's on the trilinos/master branch, I'm not sure exactly when that will happen. It's an automated process which seems to be failing the last few nights.

**Comment by jrood-nrel** [comment]
I will let @sayerhs decide. I have the tests updated to reflect the current situation. So this PR should solve them. https://my.cdash.org/index.php?project=Nalu-Wind

**Comment by sayerhs** [comment]
I have restarted the CI check for this PR... if that succeeds we can merge this.

**Pull request merged**